src/WebJobs.Script/AppMonitoringService.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Reactive.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Azure.WebJobs.Script.Eventing;
+using Microsoft.Azure.WebJobs.Script.Eventing.File;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public class AppMonitoringService : IHostedService, IDisposable
```

Comment:
I thought about this a bit. The goal of this IHostedService is to monitor any WebHost / App level changes. `app_offline.htm` happens to be the only one right now. Since it monitors if the app is offline (at the app level), I called it this. It is monitoring files, but calling it `FileMonitoringService` is still confusing IMO (we already have one). 

Any recommendation for a name? I am open to suggestions :)